server/src/main/java/org/opensearch/transport/BaseInboundMessage.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.transport;
+
+/**
+ * Base class for inbound data as a message.
+ * Different implementations are used for different protocols.
+ *
+ * @opensearch.internal
+ */
+public interface BaseInboundMessage {
+
+    /**
+     * The protocol used to encode this message
+     */
+    enum Protocol {
```

Comment:
I would drop the enumeration, it inherently prohibits extensibility (the values have to be provided ahead of time), let just use a simple `String`. The `default` is probably not descriptive, let's use something like `binary` or `internal` may be

Reply:
May be `native`?

Reply:
Done.